amanuensis/resources/project/__init__.py

```diff
@@ -44,16 +44,14 @@ def create(current_session, logged_user_id, is_amanuensis_admin, name, descripti
 
 def upload_file(session, key, project_id, expires=None):
 
-    bucket = list(config['AWS_CREDENTIALS'].keys())[0]
-
     try:
-        presigned_url = flask.current_app.boto.presigned_url(bucket, key, expires, bucket, method="put_object")
+        presigned_url = flask.current_app.s3_boto.presigned_url(config["DATA_DOWNLOAD_BUCKET"], key, expires, {}, method="put_object")
```

Comment:
I don't think we are using the DATA_DOWNLOAD_BUCKET value you can just get it from AWS_CREDENTIALS, or restructure that object to contain the value from 

AWS_CREDENTIALS:
    bucket_name:
         access_key: "key"
         secret: "secret"
         
to:
AWS_CREDENTIALS:
    DATA_DELIVERY_S3_BUCKET:
         access_key: "key"
         secret: "secret"
         bucket_name: "name"
         

in the second case you would need to update cloud automation to populate this new field and the config_default to have the field

Reply:
The reason I used DATA_DOWNLOAD_BUCKET instead of adding the bucket_name key was you have to remove the bucket_name key from config[AWS_CREDENTIALS][DATA_DELIVERY_S3_BUCKET] any time you initialize BotoManager cause it throws an 'unexpected_key' error. I can switch it though and work around that

Reply:
Why does it throw that error? Is that a private aws boto string? If so we can just use a different name. If it throws the error just because it doesn't exist in the default-config.json file then we can add it there.

Reply:
the error comes from aws' boto package, config[AWS_CREDENTIALS][DATA_DELIVERY_S3_BUCKET] returns a dictionary which gets unpacked and passed to boto as key pairs and since boto.client function does not take any parameter related to buckets an unexpected keyword error is thrown, 

Example 
creds = {'bucket_name': "bucket_name"}
client = BotoManager(creds, logger)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/pcdc_aws_client/boto.py", line 40, in __init__
    self.s3_client = client('s3', **config)
  File "boto3/__init__.py", line 92, in client
    return _get_default_session().client(*args, **kwargs)
TypeError: client() got an unexpected keyword argument 'bucket_name'

Reply:
when passing it to boto I would do a copy by value of `config[AWS_CREDENTIALS][DATA_DELIVERY_S3_BUCKET]` and a `del new_variable["bucket_name"]. 
I prefer keeping things organized in the config and adding a little business logic than spreading everything around.


Another option is to nest it one more layer:
AWS_CREDENTIALS -> DATA_DELIVERY_S3_BUCKET -> access_key, secret_key
and
AWS_CONFIGURATIONS -> DATA_DELIVERY_S3_BUCKET -> bucket-name and others 

maintaining the same key DATA_DELIVERY_S3_BUCKET to easily match them.

Reply:
that sounds good with me I can do that

Reply:
I added the bucket name to this part AWS_CREDENTIALS -> DATA_DELIVERY_S3_BUCKET and then have it removed when creating the aws client